simhash/__init__.py

```diff
@@ -69,12 +69,25 @@ def build_by_text(self, content):
         return self.build_by_features(features)
 
     def build_by_features(self, features):
-        hashs = [self.hashfunc(w.encode('utf-8')) for w in features]
+        """
+        `features` might be a list of unweighted tokens (a weight of 1
+                   will be assumed), a list of (token, weight) tuples or
+                   a token -> weight dict.
+        """
         v = [0] * self.f
         masks = [1 << i for i in range(self.f)]
-        for h in hashs:
+        if isinstance(features, dict):
+            features = features.items()
+        for f in features:
+            if isinstance(f, collections.Iterable) and \
+               not isinstance(f, basestring):
```

Comment:
Why not just check if `f` is `basestring`? What's the reason to check if `f` is `collections.Iterable`?